### PR TITLE
Fix(InputItem): add try catch to getSelection (#3236)

### PR DESCRIPTION
**Pull request by zack24q**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3237)

### CODE REVIEW

File: components/input-item/index.tsx [hunk 1]
```diff
@@ -92,7 +92,15 @@ class InputItem extends React.Component<InputItemProps, any> {
 
   onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const el = e.target;
-    const { value: rawVal, selectionEnd: prePos } = el;
+    const { value: rawVal } = el;
+
+    let prePos = 0;
+    try {
+      prePos = el.selectionEnd || 0;
```
Comment (likezero) [inline components/input-item/index.tsx]: 备注下为啥会异常？

Comment (likezero) [reply components/input-item/index.tsx]: https://github.com/ant-design/ant-design-mobile/commit/05b9d8636b5d8148d2b651f445c0e1f320629e0b

Comment (likezero) [reply components/input-item/index.tsx]: 感觉这样写会比较直观。

```js
try {
  
      prePos = el.selectionEnd;
} catch (e) {
   prePos = 0;
}
```

Comment (zack24q) [reply components/input-item/index.tsx]: 这样子改TS会报错：不能将类型“number | null”分配给类型“number”

Comment (zack24q) [reply components/input-item/index.tsx]: 异常的原因注释已加哈

Comment (ziluo) [reply components/input-item/index.tsx]: catch到错误只是`console.warn`么，没有兜底措施吗？

Comment (zack24q) [reply components/input-item/index.tsx]: 兜底就是不处理光标位置了

**Comment by doxiaodong** [comment]
这个问题改了好几次了的样子。。

**Pull request merged**